test/HHConfiguration.py

```diff
@@ -52,22 +49,24 @@
             minDR_l_j_Cut = cms.untracked.double(0.3),
             hltDRCut = cms.untracked.double(0.3),
             hltDPtCut = cms.untracked.double(0.5)  # cut will be DPt/Pt < hltDPtCut
-            ),
+            )
         )
     )
+
+# Add PUPPI MET
+puppiCfg = cms.PSet(METProducer.default_configuration.clone())
+puppiCfg.prefix = cms.string('puppimet_')
+puppiCfg.parameters.met = cms.untracked.InputTag('slimmedMETsPuppi')
+framework.addProducer('puppimet', puppiCfg)
+
 #framework.redoJEC()
+framework.smearJets()
 framework.doSystematics(['jec', 'jer'])
 process = framework.create()
 
-# Add PUPPI MET
-process.framework.producers.puppimet = cms.PSet(METProducer.default_configuration.clone())
-process.framework.producers.puppimet.prefix = cms.string('puppimet_')
-process.framework.producers.puppimet.parameters.met = cms.untracked.InputTag('slimmedMETsPuppi')
-
-# Custom the cuts
+# If we want to modify fwk parameters, it has to be done after create()
```

Comment:
Nop, it'd throw an error if you try. The `framework` object is frozen after a call to the `create` function. What this comments means is that, if you want to tweak some producers, since there's (currently) no access point from the `Framework` class, you have to pass by the `process`, which is only available after calling `create`.

But, if you do systematics, producers have already been cloned to run on systematics collection, so if you modify the `jets` producer, you also have to modify the `jets_jecup`, ... producers to have something consistent.

We need to provide a entry point in the `Framework` class to tweak producers parameters to avoid this situation.